### ISSUE TITLE
False positive for E0081 (duplicated enum discriminants)

### ISSUE THREAD

**Issue by lovasoa** [intellij-rust/intellij-rust#4117]
## Environment

* **IntelliJ Rust plugin version:** 0.2.101.2129-191
* **Rust toolchain version:** 1.36.0 (a53f9df32 2019-07-03) x86_64-apple-darwin
* **IDE name and version:** IC-191.7479.19
* **Operating system:** macOS 10.14.5

## Problem description
When an enum variant has a discriminant that is not a literal, intellij-rust always thinks that the discriminant is 0, and thus may warn that a discriminant is used twice ([E0081](https://doc.rust-lang.org/error-index.html#E0081)).

## Steps to reproduce
```rust
#[repr(u32)]
enum Dx {
    LEFT = std::u32::MAX - 1,
    MIDDLE = 0,
    RIGHT = 1,
}
```      
[playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=590bdd397528e559b22c0e1559ce70b1)

**Comment by lovasoa** [comment]
The same thing seems to happen when using negative literals : 
![image](https://user-images.githubusercontent.com/552629/60889025-5e094b80-a258-11e9-8173-51841f13cb84.png)

**Comment by Undin** [comment]
Thanks for the report!
By the way, did you use `Create New Issue` action to open this issue or fill environment info manually?

**Comment by lovasoa** [comment]
Yes, I used the new *Create New Issue* feature. It is very handy !
